app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -167,6 +167,24 @@ class AgendaViewModel @Inject constructor(
 
     }
 
+    fun onSwipeRefresh() {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
```

Comment:
Switching the context should always be the job of the suspend function and not of the coroutine that calls it. So in this case, it should happen in `syncAgenda()` and `updateLocalAgendaDayFromRemote()` (or in the suspend functions these functions use like Room queries or Retrofit calls which already do this).
In general a suspend function should always be main-safe, meaning it's safe to call from the main-thread

Reply:
ah ok , this makes more sense to me